tests/perforce_utils.py

```diff
@@ -159,12 +159,12 @@ def perforce_connection(request, docker_perforce):
     p4.password = docker_perforce.password
     p4.connect()
     p4.run_login()
-    yield p4
+    yield utils.Params(p4=p4, password=docker_perforce.password)
     p4.disconnect()
 
 
 class PerforceWorkspace(utils.BaseVcsClient):
-    def __init__(self, connection: P4, directory: py.path.local):
+    def __init__(self, connection: P4, password: str, directory: py.path.local):
```

Comment:
Please remove the password from the PerforceWorkspace.
It should only provide our "connection" class with password, not store the password on its own.